specification/appendix/commitment_discounts.md

```diff
@@ -0,0 +1,278 @@
+
+```json
+[
+    {
+        "BillingPeriodStartDate": "2023-01-01 00:00:00Z",
+        "BillingPeriodEndDate": "2023-02-01 00:00:00Z",
+        "ChargePeriodStartDate": "2023-01-01 00:00:00Z",
+        "ChargePeriodEndDate": "2023-01-01 01:00:00Z",
+        "ChargeCategory": "Purchase",
+        "ChargeFrequency": "Recurring",
+        "PricingCategory": "Committed",
+        "ResourceId": "<commitment-discount-id>",
+        "BilledCost": 1.00,
+        "EffectiveCost": 0.00,
+        "CommitmentDiscountId": "<commitment-discount-id>",
+        "CommitmentDiscountPurchasedQuantity": 1.00,
+        "CommitmentDiscountUnit": "USD"
+    },
+
+    /* ... 743 more recurring purchase records for the billing period ... */
+]
+```
+
+#### Scenario #3: Hybrid
+
+Half of the commitment is billed _once_ during the first charge period of the term for $4,380 (derived as `24 hours * 182.5 days * $1.00`), and the other half is billed across each charge period over the term, derived as (`$1.00 * 8,760 hours * 0.5`). Amortized costs incur half of the amount (i.e. $0.50) from the upfront purchase and the other half from the recurring purchase.
+
+```json
+[
+    {
+        "BillingPeriodStartDate": "2023-01-01 00:00:00Z",
+        "BillingPeriodEndDate": "2023-02-01 00:00:00Z",
+        "ChargePeriodStartDate": "2023-01-01 00:00:00Z",
+        "ChargePeriodEndDate": "2024-01-01 00:00:00Z",
+        "ChargeCategory": "Purchase",
+        "ChargeFrequency": "One-Time",
+        "PricingCategory": "Committed",
+        "ResourceId": "<commitment-discount-id>",
+        "BilledCost": 4380.00,
+        "EffectiveCost": 0.00,
+        "CommitmentDiscountId": "<commitment-discount-id>",
+        "CommitmentDiscountPurchasedQuantity": 4380.00,
+        "CommitmentDiscountUnit": "USD"
+    },
+    {
+        "BillingPeriodStartDate": "2023-01-01 00:00:00Z",
+        "BillingPeriodEndDate": "2023-02-01 00:00:00Z",
+        "ChargePeriodStartDate": "2023-01-01 00:00:00Z",
+        "ChargePeriodEndDate": "2023-01-01 01:00:00Z",
+        "ChargeCategory": "Purchase",
+        "ChargeFrequency": "Recurring",
+        "PricingCategory": "Committed",
+        "ResourceId": "<commitment-discount-id>",
+        "BilledCost": 0.50,
+        "EffectiveCost": 0.00,
+        "CommitmentDiscountId": "<commitment-discount-id>",
+        "CommitmentDiscountPurchasedQuantity": 0.50,
+        "CommitmentDiscountUnit": "USD"
+    },
+
+    /* ... 743 more recurring purchase records for the billing period ... */
+]
+```
+
+### Usage Rows
+
+Amortization of commitment discounts occur the same way regardless of how one or more purchases are made.  The same usage-based or spend-based amount is applied evenly across all charge periods and potentially allocated to eligible resources.  Continuing with the same commitment discount example, a one-year, spend-based commitment discount with a $1.00 hourly commitment, 4 types of scenarios can occur during a charge period:
```

Comment:
This section should remind the reader that there's only one resource in play in the example.